SparkPost/2web2ui
# SA-1712: Update billing UI for Saved Reports

### What Changed
 - Added conditions to the features comparison.
 - Added saved reports to the feature comparison.
 - Added saved reports to the plan change check.
 - Added lots of tests.

### How To Test
 - Go to an account with saved reports enabled. Go to `/account/billing/plan`.
 - Click on `compare features` and verify that the modal opens and that you can see a row for saved reports.
 - If the account has >30 saved reports & is on premium (appteam stg works), change plans to starter. Check that there is an authorization message that prevents the user from switching. Then switch to free and verify that the message is not present then.
 - Go to an account without saved reports enabled. Go to `/account/billing/plan`.
 - Click on `compare features` and verify that the modal opens and there is no row for saved reports.
 - Try changing plans and verify that there is no message about saved reports.
